tests/app/service/test_rest.py

```diff
@@ -1475,6 +1467,9 @@ def test_get_detailed_services_for_date_range(
     assert data[0]['statistics'][LETTER_TYPE] == {'delivered': 0, 'failed': 0, 'requested': 0}
 
     # Teardown
+    notify_db_session.session.execute(
+        delete(FactNotificationStatus).where(FactNotificationStatus.service_id == template.service.id)
+    )
```

Comment:
The above asserts should be in a "try", and this teardown should be in a "finally".  This ensures the teardown runs when the test fails.